src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/EndUpdateInstance.sql

```diff
@@ -0,0 +1,92 @@
+﻿/*************************************************************
+    Stored procedures for updating an instance status.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     EndUpdateInstance
+--
+-- DESCRIPTION
+--     Bulk update all instances in a study, creates new entry in changefeed.
+--
+-- PARAMETERS
+--     @batchSize
+--         * The desired number of instances per batch. Actual number may be smaller.
+--     @partitionKey
+--         * The partition key.
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @patientId
+--         * The Id of the patient.
+--     @patientName
+--         * The name of the patient.
+--     @patientBirthDate
+--         * The patient's birth date.
+--
+-- RETURN VALUE
+--     None
+--
+CREATE OR ALTER PROCEDURE dbo.EndUpdateInstance
+    @batchSize                          INT,
+    @partitionKey                       INT,
+    @studyInstanceUid                   VARCHAR(64),
+    @patientId                          NVARCHAR(64) = NULL,
+    @patientName                        NVARCHAR(325) = NULL,
+    @patientBirthDate                   DATE = NULL
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @rowsUpdated INT = 0
+    DECLARE @imageResourceType AS TINYINT = 0
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
+    DECLARE @updatedInstances AS TABLE
+           (PartitionKey INT,
+            StudyInstanceUid VARCHAR(64),
+            SeriesInstanceUid VARCHAR(64),
+            SopInstanceUid VARCHAR(64),
+            StudyKey BIGINT,
+            SeriesKey BIGINT,
+            InstanceKey BIGINT,
+            Watermark BIGINT)
+
+    DECLARE @totalCount INT = (SELECT COUNT(*) FROM dbo.Instance WHERE PartitionKey = @partitionKey AND StudyInstanceUid = @studyInstanceUid AND Status = 1 AND NewWatermark IS NOT NULL) 
+
+    WHILE (@rowsUpdated < @totalCount)
```

Comment:
How does the batch update help here? From my understanding since you are not committing the transaction you are still holding the lock and there is no need for this while loop

Reply:
The batching is not helping, do you suggest to just update all the rows at once. I was always under the impression it is going to be too much.

Reply:
Removed batching